openpathsampling/analysis/network.py

```diff
@@ -305,15 +312,245 @@ def rate_matrix(self, storage, force=False):
         return self._rate_matrix
 
 
-
-
 #def multiple_set_minus_switching(mistis, storage):
 
 class MISTISNetwork(TISNetwork):
-    def __init__(self, transitions):
+    """
+    Multiple interface set TIS network.
+
+    Attributes
+    ----------
+    input_transitions : list of TISTransition
+        the transitions given as input
+    """
+    # NOTE: input_transitions are in addition to the sampling_transitions
+    # and the transitions (analysis transitions)
+    def __init__(self, input_transitions):
         super(MISTISNetwork, self).__init__()
+        self.input_transitions = input_transitions
+
+        if not hasattr(self, '_sampling_transitions'):
+            self.build_sampling_transitions(input_transitions)
+
+        self.build_analysis_transitions()
+
+        if not hasattr(self, 'movers'):
+            self.movers = {}
+            self.build_movers()
+
+
+    def to_dict(self):
+        ret_dict = {
+            'movers' : self.movers,
+            'minus_ensembles' : self.minus_ensembles,
+            'ms_outers' : self.ms_outers,
+            'transition_pairs' : self.transition_pairs,
+            'input_transitions' : self.input_transitions
+        }
+        return ret_dict
+
+    @staticmethod
+    def from_dict(dct):
+        network = MISTISNetwork.__new__(MISTISNetwork)
+        network.movers = dct['movers']
+        network.minus_ensembles = dct['minus_ensembles']
+        network.ms_outers = dct['ms_outers']
+        network.transition_pairs = dct['transition_pairs']
+        network.__init__(dct['transitions'])
```

Comment:
Quick question here before merging. Is it possible that calling the `__init__` at the end that the restored movers, ensembles, etc get overridden by newly created ones? If so, this should be used:
```
network = MISTISNetwork(dct['transitions'])
network.movers = dct['movers']
network.minus_ensembles = dct['minus_ensembles']
network.ms_outers = dct['ms_outers']
network.transition_pairs = dct['transition_pairs']
```